module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+#include <iomanip>
+
+void printMap (std::array<std::array<uint8_t, width>, height> bitmap)
+{
+    for(const auto& line : bitmap)
+    {
+        for (const auto c : line)
+        {
+            std::cout<<std::setw(3)<<(int)c<<' ';
+
+        }
+        std::cout<<'\n';
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale (std::array<std::array<uint8_t, width>, height> bitmap)
```

Comment:
const ref 
```suggestion
std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale (const std::array<std::array<uint8_t, width>, height>& bitmap)
```

Reply:
If you do not modify value it's good idead to pass it as const ref to optimize program and don't create needless copy.